### PR TITLE
Leak fixes 

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: winpr/libwinpr/file/file.c [hunk 1]
```diff
@@ -253,7 +253,7 @@ static BOOL FileLockFileEx(HANDLE hFile, DWORD dwFlags, DWORD dwReserved,
 		DWORD nNumberOfBytesToLockLow, DWORD nNumberOfBytesToLockHigh,
 		LPOVERLAPPED lpOverlapped)
  {
-	int lock;
+	int lock = 0;
```
Comment (hardening) [inline winpr/libwinpr/file/file.c]: lock is always set a few lines below, so I don't think the initialization is useful

**Comment by hardening** [comment]
+1 (non-blocking remark)

File: winpr/libwinpr/file/file.c (hunk 1 above)
Comment (akallabeth) [inline winpr/libwinpr/file/file.c]: @hardening Yes, did one too much ^^

File: winpr/libwinpr/file/file.c [hunk 2]
```diff
@@ -424,7 +424,7 @@ static HANDLE FileCreateFileA(LPCSTR lpFileName, DWORD dwDesiredAccess, DWORD dw
 	WINPR_FILE* pFile;
 	BOOL create;
 	const char* mode = FileGetMode(dwDesiredAccess, dwCreationDisposition, &create);
-	int lock;
```
Comment (mfleisz) [inline winpr/libwinpr/file/file.c]: I think this lock is also always set when it is used (either if FILE_SHARE_READ or FILE_SHARE_WRITE is set in dwShareMode)

Comment (akallabeth) [inline winpr/libwinpr/file/file.c]: @mfleisz No, look again ;) it is only set if ```dwSharedMode & (FILE_SHARE_READ | FILE_SHARE_WRITE)```

**Pull request merged**